guides/plugins/plugins/api/multi-inventory.md

```diff
@@ -0,0 +1,146 @@
+    "name": "Warehouse A",
+    "groups": [{
+        "id": "8cf7736855594501aaf86351e147c61e"
+    }, {
+        "id": "4154501a3812fcb6a501aaf8c7736855"
+    }]
+}
+```
+
+### Assigning WarehouseGroups to Products, creating ProductWarehouses via assocation
+
+```json
+
+// POST /api/_action/sync
+
+[{
+    "action": "upsert",
+    "entity": "product",
+    "payload": [{
+        "id": "86d38702be7e4ac9a941583933a1c6f5",
+        "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
+        "warehouseGroups": [{
+            "id": "8cf7736855594501aaf86351e147c61e"
+        }],
+        "warehouses": [{
+            "id": "f5c850109fe64c228377cbd369903b75",
+            "productId": "86d38702be7e4ac9a941583933a1c6f5",
+            "productVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
+            "warehouseId":"4ce2bd36d2824153812fcb6a97f22d22",
+            "stock": 0
+        }]
+    }]
+}]
+```
+
+### Updating ProductWarehouse stocks
+
+You can update `product_warehouse.stock` in batch via SyncApi, or patch a specific entity directly via entity repository.
+
+```json
+
+// POST /api/_action/sync
+
+[{
+    "action": "upsert",
+    "entity": "product_warehouse",
+    "payload": [{
+        "id": "f5c850109fe64c228377cbd369903b75",
+        "stock": 1500
+    }, {
+        "id": "228377cbd369903b75f5c850109fe64c",
+        "stock": 0
+    }]
+}]
+
+// PATCH /api/product-warehouse/f5c850109fe64c228377cbd369903b75
+
+{
+    "id": "f5c850109fe64c228377cbd369903b75",
+    "stock": 1500
+}
+```
+
+## Concept
+
+Note: Every described behavior only applies to Products, which are assigned to WarehouseGroups - every unrelated Product will use the default Shopware behavior.
+
+### ERP System as Single-Source-of-Truth
+
+MultiInventory is intended to be used as an interface between Shopware and your resource management software. This means that Shopware will only calculate availability of Products based on your Warehouse configuration, and only changes stocks of a Warehouse, when an order is created. This is to prevent oversales, while also making your system the single source of truth - making it easier to maintain both systems at the same time.
+
+### Product Availability
+
+Availability of Products is defined in 2 steps
+
+* WarehouseGroups can be assigned to Rules (Rule builder)
+  * If the rule is invalid, this Group will not be considered in calculating Product availability.
+  * Products / Warehouses still can be available via other groups.
+* Products can have a stock per Warehouse
+  * All Warehouses inside an active WarehouseGroup are taken into account for calculating the total stock of a specific Product.
+  * Warehouses are unique, but can be assigned to multiple Groups (e.g. all Warehouses in the Group "Germany" can also be in the Group "Europe").
+
+If both conditions are true (e.g. "Customer is in a specific customer group" and "requested stock <= total Product stock of all valid Warehouses"), the requested Product is considered available.
```

Comment:
The stock availability is also compatible with other product configurations like maxPurchase, minPurchase, purchaseSteps, etc.